data/data/agent/files/usr/local/bin/agent-gather

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+function gather_journal() {
+	( >&2 echo -n "Gathering node journal" )
+	journalctl -o export > "${ARTIFACTS_DIR}/journal.export"
+	( >&2 echo " Done")
+}
+
+function gather_agent_data() {
+	( >&2 echo -n "Gathering agent installation data" )
+	mkdir -p "${ARTIFACTS_DIR}/etc"
+	cp -a /etc/assisted{,-service} "${ARTIFACTS_DIR}/etc/"
+	( >&2 echo -n ".")
+	mkdir "${ARTIFACTS_DIR}/etc/containers"
+	cp -a /etc/containers/registries.conf "${ARTIFACTS_DIR}/etc/containers/"
+	( >&2 echo -n ".")
+	mkdir -p "${ARTIFACTS_DIR}/usr/local/share/"
+	cp -a /usr/local/share/assisted-service "${ARTIFACTS_DIR}/usr/local/share/"
+
+	if [[ $(podman info -f json | jq 'if any(.registries[]; type=="array") then false else any(.registries[].Mirrors[]; has("Location")) end') == "true" ]]; then
+		mkdir -p "${ARTIFACTS_DIR}/etc/pki/ca-trust/extracted/"
+		cp -a /etc/pki/ca-trust/extracted/pem "${ARTIFACTS_DIR}/etc/pki/ca-trust/extracted/"
+	fi
+	( >&2 echo " Done")
+}
+
+function gather_network_data() {
+	( >&2 echo -n "Gathering network data" )
+	ip -d -j -p addr show > "${ARTIFACTS_DIR}/ipaddr"
+	( >&2 echo -n ".")
+        ip -d -j -p link show > "${ARTIFACTS_DIR}/iplink"
+	( >&2 echo -n ".")
+        ip -d -j -p route show > "${ARTIFACTS_DIR}/iproute"
+	( >&2 echo -n ".")
+	cp /etc/resolv.conf "${ARTIFACTS_DIR}/resolv.conf"
+	( >&2 echo " Done")
+}
+
+function gather_storage_data() {
+	( >&2 echo  -n "Gathering storage data" )
+	mkdir -p "${ARTIFACTS_DIR}/etc"
+	cp /etc/mtab "${ARTIFACTS_DIR}/etc/mtab"
+	( >&2 echo -n ".")
+	lsblk > "${ARTIFACTS_DIR}/lsblk"
+	( >&2 echo " Done")
+}
+
+function Help()
+{
+	echo "Gathers the necessary data for troubleshooting OpenShift's agent based installation"
+	echo
+	echo "Syntax: agent-gather [-h|-v]"
+	echo "options:"
+        echo "-h	Print this help"
+        echo "-O	Output the compressed content to stdout"
+	echo "-v	Set verbose mode"
+	echo
+}
+
+while getopts ":hvO" option; do
+	case $option in
+		h)
+			Help
+			exit;;
+		v)
+			set -xv;;
+		O)
```

Comment:
There is an issue when -O is used:

```
[core@api ~]$ agent-gather -O
This command must be run with super user privileges. Doing that now
Gathering node journal Done
Gathering agent installation data.. Done
Gathering network data... Done
Gathering storage data. Done
Compressing gathered data to ./agent-gather-20220927-131411UTC.tar.xz
tar: Refusing to write archive contents to terminal (missing -f option?)
tar: Error is not recoverable: exiting now
```

Reply:
Tar refuses to output to stdout in the terminal, but is fine doing it to pipes or redirections: https://stackoverflow.com/a/52705174/206009